velox/connectors/hive/iceberg/tests/IcebergReadTest.cpp

```diff
@@ -831,11 +837,164 @@ TEST_F(HiveIcebergTest, testPartitionedRead) {
+
+  // Write data file with old schema (only c0).
+  std::vector<RowVectorPtr> dataVectors;
+  auto c0 = makeFlatVector<int64_t>({100, 200, 300});
+  dataVectors.push_back(makeRowVector({"c0"}, {c0}));
+
+  auto dataFilePath = TempFilePath::create();
+  writeToFile(
+      dataFilePath->getPath(), dataVectors, config_, flushPolicyFactory_);
+
+  auto icebergSplits = makeIcebergSplits(dataFilePath->getPath());
+
+  // Read with new schema (c0, c1, and c2).
+  auto plan = PlanBuilder(pool_.get())
+                  .tableScan(newRowType, {}, "", newRowType)
+                  .planNode();
+
+  // Expected result: c0 has values, c1 and c2 are NULL.
+  auto expectedC1 = makeNullConstant(TypeKind::INTEGER, 3);
+  auto expectedC2 = makeNullConstant(TypeKind::VARCHAR, 3);
+  std::vector<RowVectorPtr> expectedVectors;
+  expectedVectors.push_back(
+      makeRowVector({"c0", "c1", "c2"}, {c0, expectedC1, expectedC2}));
+
+  AssertQueryBuilder(plan, duckDbQueryRunner_)
+      .splits(icebergSplits)
+      .assertResults(expectedVectors);
+}
+
+// Test reading partition columns from Hive-migrated tables.
+// This tests the adaptColumns method handling partition columns that are not
+// stored in the data file but provided via partitionKeys map.
+// This scenario occurs when reading Hive-written data files where partition
+// column values are stored in partition metadata rather than in the data file.
+TEST_F(HiveIcebergTest, partitionColumnsFromHive) {
+  auto fileRowType = ROW({"c0", "c1"}, {BIGINT(), INTEGER()});
+  auto tableRowType =
+      ROW({"c0", "c1", "region", "year"},
+          {BIGINT(), INTEGER(), VARCHAR(), INTEGER()});
+
+  // Write data file with only non-partition columns (c0, c1).
+  std::vector<RowVectorPtr> dataVectors;
+  auto c0 = makeFlatVector<int64_t>({1, 2, 3});
+  auto c1 = makeFlatVector<int32_t>({10, 20, 30});
+  dataVectors.push_back(makeRowVector({"c0", "c1"}, {c0, c1}));
+
+  auto dataFilePath = TempFilePath::create();
+  writeToFile(
+      dataFilePath->getPath(), dataVectors, config_, flushPolicyFactory_);
+
+  // Set partition keys for region and year.
+  std::unordered_map<std::string, std::optional<std::string>> partitionKeys;
+  partitionKeys["region"] = "US";
+  partitionKeys["year"] = "2025";
+
+  auto icebergSplits =
+      makeIcebergSplits(dataFilePath->getPath(), {}, partitionKeys);
+  auto assignments = makeColumnHandles(tableRowType, {2, 3});
+
+  // Read with table schema including partition columns.
+  auto plan = PlanBuilder(pool_.get())
+                  .tableScan(tableRowType, {}, "", tableRowType, assignments)
+                  .planNode();
+
+  // Expected result: c0 and c1 from file, region and year from partition keys.
+  auto expectedRegion = makeFlatVector<std::string>({"US", "US", "US"});
+  auto expectedYear = makeFlatVector<int32_t>({2025, 2025, 2025});
+  std::vector<RowVectorPtr> expectedVectors;
+  expectedVectors.push_back(makeRowVector(
```

Comment:
consider

```
makeRowVector(tableRowType->names(), {
   makeFlatVector<std::string>({"US", "US", "US"}),
   makeFlatVector<int32_t>({2025, 2025, 2025}),
})
```

No need for expectedRegion and expectedYear variables. Also, no need to repeat column names.